src/controllers/movie.js

```diff
@@ -0,0 +1,167 @@
+const { Prisma } = require('@prisma/client');
+const prisma = require('../utils/prisma');
+const getErorCode = (e, res) => {
+    if (e instanceof Prisma.PrismaClientKnownRequestError) {
+        if (e.code === "P2002") {
+            return res.status(409).json({ error: "A movie with the provided field already exists" })
+        } else if (e.code === "P2025") {
+            return res.status(404).json({ error: "A movie with the provided id does not exists!" })
+        }
+    }
+    res.status(500).json({ error: e.code + " " + e.message })
+}
+const getAllMovies = async (req, res) => {
+
+    const { runtimeLt, runtimeGt } = req.query;
+    let query = { include: { screenings: true } };
+
+    if (runtimeLt && runtimeGt) {
+        query = {
+            where: {
+                runtimeMins: {
+                    gte: Number(runtimeGt),
+                    lte: Number(runtimeLt)
+                }
+            },
+            include: { screenings: true }
+        }
+    }
+
+    try {
+        const movies = await prisma.movie.findMany(query);
+        res.status(200).json({ movies });
+    } catch (error) {
+        return getErorCode(error, res)
+    }
+}
+const updateWithRelationValues = (req, upid) => {
+    const { title, runtimeMins, screenings } = req.body;
+    let id = 0;
+    const mappedScreen = screenings.map((screen) => {
+
+        const { movieId, screenId, startsAt } = screen;
+        if (screen.screeningsId) {
+            id = screen.screeningsId;
+        }
```

Comment:
What's the reason for having a mutable `id` outside of the map? It doesn't look like it's used anywhere but inside the map